beater/jaeger/server.go

```diff
@@ -79,15 +94,15 @@ func NewServer(logger *logp.Logger, cfg *config.Config, tracer *apm.Tracer, repo
 		srv.grpc.server = grpc.NewServer(grpcOptions...)
 		srv.grpc.listener = grpcListener
 		// TODO(simi) to add support for sampling: api_v2.RegisterSamplingManagerServer
-		api_v2.RegisterCollectorServiceServer(srv.grpc.server, grpcCollector{traceConsumer})
+		api_v2.RegisterCollectorServiceServer(srv.grpc.server, grpcCollector{auth, traceConsumer})
 	}
 	if cfg.JaegerConfig.HTTP.Enabled {
 		// TODO(axw) should the listener respect cfg.MaxConnections?
 		httpListener, err := net.Listen("tcp", cfg.JaegerConfig.HTTP.Host)
 		if err != nil {
 			return nil, err
 		}
-		httpMux, err := newHTTPMux(traceConsumer)
+		httpMux, err := newHTTPMux(auth, traceConsumer)
```

Comment:
Do we actually want to support auth handling for HTTP, since this does not support TLS? 

If so, I think the docs should be aligned, currently they only discuss auth handling for gRPC.

Reply:
Good point. I was thinking it might still be useful for folks using a service mesh, stunnel, or some other way of securing comms without specific TLS support in the server/client. Pretty tenuous though, so I'll remove it for now.

Reply:
I renamed `apm-server.jaeger.auth_tag` to `apm-server.jaeger.grpc.auth_tag`, changed the code to only check auth in gRPC.